### ISSUE TITLE
git tagging / GitHub releases

### ISSUE THREAD

**Issue by lorenzwalthert** [rstudio/reticulate#870]
I think it's standard convention to:

* create GitHub releases for CRAN releases to allow people to 'Watch Releases only' on Github and as an alternative way to to CRAN for looking up NEWS, latest features etc.
* create git tags for CRAN releases. This facilitates installing certain versions or other git related opterations, e.g. with `remotes::install_github('rstudio/reticulate@v1.16')`.

Would you consider adding git tags and GitHub releases to this repo?

**Comment by eddelbuettel** [comment]
But "standard convention" for R packages is really only what it is in _Writing R Extension_ and the _CRAN Repository Policy_.

AFAIK the word "git" does not appear there, much less "GitHub".  So "standard convention" would depend quite a bit on whose standard it is.  All that said, many larger / well visited and followed repos do this on a purely voluntary basis. So you can of course ask nicely for it as you did, but you cannot claim any Vienna or other authority made you do it.... 

If you want a simple subscription mechanism to releases I believe @daattali wrote something where you can subscribe on a per-repo basis.  I still prefer the waterhose I built with CRANberries and follow that feed...

**Comment by daattali** [comment]
I agree that it isn't  "standard convention", but some people do that. It's not uncommon to do git releases, but it certainly isn't as common for R packages as it is in other languages. If you want to subscribe to receive package updates, as @eddelbuettel  said you can follow [CRANberries](https://dirk.eddelbuettel.com/cranberries/index.html) to get a stream of all CRAN updates, or https://cranalerts.com/ to get updates for a single package

**Comment by lorenzwalthert** [comment]
Thanks guys; let's see if the maintainers of this repo want to adhere to the convention. As I pointed out, there are advantages that emerge from using git tags that CRANberries and similar services cannot make up for (such as the pinning by tag when installing with remotes). There are major repos under the rstudio GitHub org that follow this practice already, e.g: 

- rmarkdown
- blogdown
- plumber 
- gt 
- keras 
- tensorflow 

Also, it's the case for [all pinned tidyverse repos](https://github.com/tidyverse), i.e. dplyr, purr, ggplot2, magrittr, tidyr and readr and [all pinned r-lib repos](https://github.com/r-lib), i.e. devtools, usethis, testthat, roxygen2, pkgdown, httr. At least you can say it's very common for popular repos RStudio largely contributes to.

**Comment by kevinushey** [comment]
For versioned installs from CRAN, you can also just use e.g.:

```
renv::install("reticulate@1.14")
```

That said, it seems reasonable to do this moving forward.
